src/d3a/models/market/balancing.py

```diff
@@ -42,18 +42,22 @@ def __init__(self, time_slot=None, area=None, notification_listener=None, readon
 
         super().__init__(time_slot, area, notification_listener, readonly)
 
-    def offer(self, price: float, energy: float, seller: str):
+    def offer(self, price: float, energy: float, seller: str, iaaFee: bool = False):
         assert False
 
     def balancing_offer(self, price: float, energy: float,
-                        seller: str, from_agent: bool=False) -> BalancingOffer:
+                        seller: str, from_agent: bool=False,
+                        iaaFee: bool = False) -> BalancingOffer:
         if seller not in DeviceRegistry.REGISTRY.keys() and not from_agent:
             raise DeviceNotInRegistryError(f"Device {seller} "
                                            f"not in registry ({DeviceRegistry.REGISTRY}).")
         if self.readonly:
             raise MarketReadOnlyException()
         if energy == 0:
             raise InvalidOffer()
+        if iaaFee:
+            price = price * (1 + self.transfer_fee_pct / 100)
```

Comment:
Why not holding a self.transfer_fee_fraction and eliminate these hard-to-maintain divisions/multiplications by 100 ?